db/migrations/20200615121551_create_time_auction_totals_query.sql

```diff
@@ -0,0 +1,182 @@
+-- +goose Up
+CREATE TYPE api.time_bid_total AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    count BIGINT,
+    lot_start NUMERIC,
+    lot_end NUMERIC,
+    bid_amount_start NUMERIC,
+    bid_amount_end NUMERIC
+);
+
+CREATE FUNCTION api.time_flip_bid_totals(ilk_identifier TEXT, range_start TIMESTAMP, range_end TIMESTAMP, bucket_interval INTERVAL DEFAULT '1 day'::INTERVAL)
+    RETURNS SETOF api.time_bid_total AS
+$$
+WITH buckets AS (SELECT generate_series(range_start, range_end - bucket_interval, bucket_interval) AS bucket_start),
+bid_results AS (
+    SELECT bid_id,
+        MIN(block_timestamp) AS block_timestamp,
+        MAX(lot) AS lot_start,
+        MIN(lot) AS lot_end,
+        MIN(bid_amount) AS bid_amount_start,
+        MAX(bid_amount) AS bid_amount_end
+    FROM maker.bid_event
+    LEFT JOIN public.headers ON (headers.block_number = block_height)
+    WHERE ilk_identifier = time_flip_bid_totals.ilk_identifier
+    GROUP BY bid_id
+)
+SELECT buckets.bucket_start AS bucket_start,
+       buckets.bucket_start + bucket_interval AS bucket_end,
+       bucket_interval AS bucket_interval,
+       COUNT(lot_start) AS count,
+       COALESCE(SUM(lot_start), 0) AS lot_start,
+       COALESCE(SUM(lot_end), 0) AS lot_end,
+       COALESCE(SUM(bid_amount_start), 0) AS bid_amount_start,
+       COALESCE(SUM(bid_amount_end), 0) AS bid_amount_end
+FROM buckets
+    LEFT JOIN bid_results ON (
+        block_timestamp >= extract(epoch FROM buckets.bucket_start) AND
```

Comment:
I'm a little concerned about the performance here - I don't think postgres can use any index on bucket_start once you call extract on it. It would be worth taking a look at the explain plan.

Reply:
I think you're right on this. I'll change it to precompute the epoch.